proposals/vocabularies.md

```diff
@@ -0,0 +1,267 @@
+    {"$ref": "meta/vocab2"},  // https://example.org/meta/vocab2
+    {"$ref": "meta/vocab3"}   // https://example.org/meta/vocab3
+  }
+  // ...
+}
+```
+
+Finally, the keywords in both the Core and Validation specifications will be
+divided into multiple vocabularies. The keyword definitions will be removed from
+the meta-schema and added to vocabulary schemas to which the meta-schema will
+contain references. In this way, the meta-schema's functionality remains the same.
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/next/schema",
+  "$id": "https://json-schema.org/draft/next/schema",
+  "$vocabulary": {
+    "https://json-schema.org/draft/next/vocab/core": true,
+    "https://json-schema.org/draft/next/vocab/applicator": true,
+    "https://json-schema.org/draft/next/vocab/unevaluated": true,
+    "https://json-schema.org/draft/next/vocab/validation": true,
+    "https://json-schema.org/draft/next/vocab/meta-data": true,
+    "https://json-schema.org/draft/next/vocab/format-annotation": true,
+    "https://json-schema.org/draft/next/vocab/content": true
+  },
+  "$dynamicAnchor": "meta",
+
+  "title": "Core and Validation specifications meta-schema",
+  "allOf": [
+    {"$ref": "meta/core"},
+    {"$ref": "meta/applicator"},
+    {"$ref": "meta/unevaluated"},
+    {"$ref": "meta/validation"},
+    {"$ref": "meta/meta-data"},
+    {"$ref": "meta/format-annotation"},
+    {"$ref": "meta/content"}
+  ],
+}
+```
+
+The division of keywords among the vocabularies will be in accordance with the
+2020-12 specification (for now).
+
+### Limitations
+
+#### Unknown Keywords and Unsupported Vocabularies
+
+This proposal, in its current state, seeks to mimic the behavior defined in the
+2020-12 specification. However, the current specification's disallowance of
+unknown keywords presents a problem for schemas that use keywords from optional
+vocabularies. (This is the topic of the discussion at
+https://github.com/orgs/json-schema-org/discussions/342.)
+
+In short, if a schema uses a keyword from an unknown _optional_ vocabulary, the
+implementation cannot proceed because unknown keywords are explicitly
+disallowed. However, not being able to proceed with evaluation is the behavior
+prescribed for _required_ vocabularies. Thus, if the behaviors for required and
+optional vocabularies is the same, then the boolean value is moot, which
+highlights that the structure of `$vocabulary` needs to be reconsidered.
+
+#### Machine Readability
+
+The vocabulary URI is an opaque value. There is no data that an implementation
+can reference to identify the keywords defined by the vocabulary. The vocabulary
+schema _implies_ this, but scanning a `properties` keyword isn't very reliable.
+Moreover, such a system cannot provide metadata about the keywords. As such, the
+user must explicitly ensure that the implementation recognizes and supports the
+vocabulary, which isn't much of an improvement over the current state.
```

Comment:
That statement doesn't make sense to me. I don't see how the vocabulary description document has any effect for users other than reducing the work they need to do when adding support for their vocabulary in the library they're using.

Reply:
There was always an intent to build in some kind of machine-readable vocbaulary meta-data document.  What stalled that is that we couldn't identify what that meta-data should be.

I'm merely mentioning that the current solution doesn't address that intent, and whatever we come up with should.  I think it's fine to keep in here for now.

The last sentence is just stating that the current state of having to check if a vocab is supported is no different than having to check if explicit keywords are supported.  Having a machine-readable description document could automate some of that in some cases and ease the burden of the user.

Reply:
I'm not disputing that vocab description document wasn't always the intention or that it doesn't have value. It's very valuable and I'm 100% on board with having one. I see it enabling all kinds of things. It's just the last sentence that doesn't make sense to me. I don't see how a vocabulary description document can influence the need for a user to check if an implementation supports a vocabulary or not.

Reply:
The idea was that an implementation could potentially "read in" a vocabulary and support it dynamically.  I think in practice this would be limited to vocabs which only defined annotation-only keywords, but that was the thought.

Reply:
I agree with that completely, but that seems like it's a different thing than what that sentence is saying. Maybe I may have misunderstood what you meant with the word "ensure". I read that to mean "check the documentation" or "run a tests to see if it works as expecte". But, if you meant something more like "write the code to configure the vocabulary", then the sentence makes sense to me.

Reply:
I read that to mean "check the documentation" or "run a tests to see if it works as expecte[d]". But, if you meant something more like "write the code to configure the vocabulary", then the sentence makes sense to me.

It's encompassing of both, really.  Definitely the former, but also up to and including the latter.